src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/shared-module/components/browser-base/browser-base.component.ts

```diff
@@ -196,28 +196,33 @@ export abstract class BaseBrowserComponent<TEntity extends IEntityBase> extends
      */
     public itemClicked(event: any, index: number): void {
         const itemModel = this.items[index];
-        if (event instanceof MouseEvent) {
-            if (event.ctrlKey || event.type === 'tap') {
-                this.lastSelectedIndex = index;
-                this.selectionService.toggle(itemModel.item);
-                return;
-            } else if (event.shiftKey) {
-                const firstIndex = this.lastSelectedIndex < index ? this.lastSelectedIndex : index;
-                const lastIndex = this.lastSelectedIndex < index ? index : this.lastSelectedIndex;
-
-                const itemsToAdd = this.items.slice(firstIndex, lastIndex + 1).map(i => i.item);
-
-                this.selectionService.clear();
-                this.selectionService.addItems(itemsToAdd);
-                return;
-            }
+        if (event.ctrlKey || event.type === 'tap') {
+            this.lastSelectedIndex = index;
+            this.selectionService.toggle(itemModel.item);
+            return;
+        } else if (event.shiftKey) {
+            const firstIndex = this.lastSelectedIndex < index ? this.lastSelectedIndex : index;
+            const lastIndex = this.lastSelectedIndex < index ? index : this.lastSelectedIndex;
+
+            const itemsToAdd = this.items.slice(firstIndex, lastIndex + 1).map(i => i.item);
+
+            this.selectionService.clear();
+            this.selectionService.addItems(itemsToAdd);
+            return;
         }
 
         this.lastSelectedIndex = index;
         this.selectionService.clear();
         this.selectionService.addItem(itemModel.item);
     }
 
+    public itemPressed(index: number): void {
+        const itemModel = this.items[index];
+        this.lastSelectedIndex = index;
+        this.selectionService.clear();
```

Comment:
Should we clear the selected items or do a toggle instead? 
Imagining we are on mobile and I want to select multiple times, I don't know how we could achieve that if we always clear the selected items.

Reply:
Move the tap event handler from the item clicked to the item pressed